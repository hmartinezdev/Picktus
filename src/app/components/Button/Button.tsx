import colors from '@constants/colors';
import { borderRadius, fontFamily } from '@constants/styles';
import React, { Component } from 'react';

export interface IButtonPropTypes {
  text?: string;
  children?: JSX.Element[] | JSX.Element;
  onClick(): void;
}

class Button extends Component<IButtonPropTypes, {}> {
  public static defaultProps = {
    text: '',
  };

  public render(): React.ReactElement<Button> {
    const { onClick, text, children } = this.props;

    return (
      <button onClick={onClick} className="button">
        {children ? children : <p className="text">{text}</p>}
        <style jsx>{`
          .button {
            display: flex;
            justify-content: center;
            align-items: center;
            color: ${colors.white};
            transition: all 0.5s;
            position: relative;
            text-align: center;
            cursor: pointer;
            font-family: ${fontFamily};
            border-radius: ${borderRadius};
            padding: 0 0.7rem;
            width: 75%;
            height: 2.35rem;
            min-width: 28%;
            margin: 0.7rem 0;
            background: none;
            border: none;
            outline: none;
          }

          :global(.button svg) {
            fill: ${colors.white};
            transition: all 0.5s;
          }

          .button:hover {
            color: ${colors.blue};
          }

          :global(.button:hover svg) {
            fill: ${colors.blue};
          }

          .text {
            line-height: 2.35rem;
            font-size: 0.75rem;
            z-index: 2;
          }

          .button:active::after {
            background-color: ${colors.green}ca;
          }

          .button a {
            color: rgba(51, 51, 51, 1);
            text-decoration: none;
            display: block;
            cursor: pointer;
          }

          .button::before {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            background-color: ${colors.blue}45;
            transition: all 0.3s;
            border-radius: ${borderRadius};
            cursor: pointer;
          }
          .button:hover::before {
            opacity: 0;
            transform: scale(0.5, 0.5);
          }
          .button::after {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            opacity: 0;
            transition: all 0.3s;
            border: 1px solid ${colors.lightGreen}ca;
            transform: scale(0.7, 0.7);
            border-radius: ${borderRadius};
            cursor: pointer;
          }

          .button:hover::after {
            opacity: 1;
            transform: scale(1, 1);
          }
        `}</style>
      </button>
    );
  }
}

export default Button;
