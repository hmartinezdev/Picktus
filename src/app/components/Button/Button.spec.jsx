import { mount, shallow } from 'enzyme';
import React from 'react';
import Button, { ButtonPropTypes } from './Button';

let wrapper;
const baseProps = { onClick: () => undefined };
const setup = (props = baseProps) => shallow(<Button {...props} />);
const mountSetup = (props = baseProps) => mount(<Button {...props} />);

describe('<Button />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should put the text props inside the button', () => {
    wrapper = setup({ onClick: () => undefined, text: 'test' });
    expect(wrapper).toMatchSnapshot();
  });

  test('should put the onClick props on the button', () => {
    wrapper = setup({ onClick: () => undefined });
    expect(wrapper).toMatchSnapshot();
  });

  test('click on the button should call the onClick props', () => {
    const onClick = jest.fn();
    wrapper = setup({ onClick });
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
