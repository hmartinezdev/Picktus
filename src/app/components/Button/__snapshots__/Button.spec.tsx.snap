// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should put the onClick props on the button 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  <span />
  <style
    jsx={true}
  >
    
          .button {
            color: #F8F8F7;
            transition: all 0.5s;
            position: relative;
            text-align: center;
            cursor: pointer;
            font-family: 'Josefin Sans', sans-serif;
            boder-radius: 3px;
            padding: 0.8rem 0.7rem 0.8rem;
            font-size: 0.75rem;
            width: 75%;
            margin: 0.5rem 0;
            background: none;
            border: none;
            outline: none;
          }

          .button:active::after {
            background-color: #75ab88ca;
          }

          .button a {
            color: rgba(51, 51, 51, 1);
            text-decoration: none;
            display: block;
            cursor: pointer;
          }

          .button::before {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            background-color: #9fcbe745;
            transition: all 0.3s;
            boder-radius: 3px;
            cursor:pointer;
          }
          .button:hover::before {
            opacity: 0;
            transform: scale(0.5, 0.5);
          }
          .button::after {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            opacity: 0;
            transition: all 0.3s;
            border: 1px solid #75ab88ca;
            transform: scale(0.7, 0.7);
            boder-radius: 3px;
            cursor: pointer;
          }
          .button:hover::after {
            opacity: 1;
            transform: scale(1, 1);
          }
        
  </style>
</button>
`;

exports[`<Button /> should put the text props inside the button 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  <span>
    test
  </span>
  <style
    jsx={true}
  >
    
          .button {
            color: #F8F8F7;
            transition: all 0.5s;
            position: relative;
            text-align: center;
            cursor: pointer;
            font-family: 'Josefin Sans', sans-serif;
            boder-radius: 3px;
            padding: 0.8rem 0.7rem 0.8rem;
            font-size: 0.75rem;
            width: 75%;
            margin: 0.5rem 0;
            background: none;
            border: none;
            outline: none;
          }

          .button:active::after {
            background-color: #75ab88ca;
          }

          .button a {
            color: rgba(51, 51, 51, 1);
            text-decoration: none;
            display: block;
            cursor: pointer;
          }

          .button::before {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            background-color: #9fcbe745;
            transition: all 0.3s;
            boder-radius: 3px;
            cursor:pointer;
          }
          .button:hover::before {
            opacity: 0;
            transform: scale(0.5, 0.5);
          }
          .button::after {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            opacity: 0;
            transition: all 0.3s;
            border: 1px solid #75ab88ca;
            transform: scale(0.7, 0.7);
            boder-radius: 3px;
            cursor: pointer;
          }
          .button:hover::after {
            opacity: 1;
            transform: scale(1, 1);
          }
        
  </style>
</button>
`;

exports[`<Button /> should render properly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  <span />
  <style
    jsx={true}
  >
    
          .button {
            color: #F8F8F7;
            transition: all 0.5s;
            position: relative;
            text-align: center;
            cursor: pointer;
            font-family: 'Josefin Sans', sans-serif;
            boder-radius: 3px;
            padding: 0.8rem 0.7rem 0.8rem;
            font-size: 0.75rem;
            width: 75%;
            margin: 0.5rem 0;
            background: none;
            border: none;
            outline: none;
          }

          .button:active::after {
            background-color: #75ab88ca;
          }

          .button a {
            color: rgba(51, 51, 51, 1);
            text-decoration: none;
            display: block;
            cursor: pointer;
          }

          .button::before {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            background-color: #9fcbe745;
            transition: all 0.3s;
            boder-radius: 3px;
            cursor:pointer;
          }
          .button:hover::before {
            opacity: 0;
            transform: scale(0.5, 0.5);
          }
          .button::after {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            z-index: 1;
            opacity: 0;
            transition: all 0.3s;
            border: 1px solid #75ab88ca;
            transform: scale(0.7, 0.7);
            boder-radius: 3px;
            cursor: pointer;
          }
          .button:hover::after {
            opacity: 1;
            transform: scale(1, 1);
          }
        
  </style>
</button>
`;
